### PR TITLE
fix: On app exit, close session as healthy

### PR DESCRIPTION
**Pull request by bruno-garcia**
## :loudspeaker: Type of change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description

Handle on close notification to close session as Exited

## :green_heart: How did you test it?
On device

## :pencil: Checklist

- [ ] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] All tests passing
- [x] I've updated the CHANGELOG

### CODE REVIEW

**Comment by eric** [comment]
Is `UIApplicationWillTerminateNotification` called for apps that don't do any background execution?

**Comment by bruno-garcia** [comment]
@eric I'm not sure I follow.  What do you mean by "any background" execution?

Based on Apple's docs this notification is: _Posted when the app is about to terminate._

**Comment by eric** [comment]
The [documentation for `applicationWillTerminate:`](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1623111-applicationwillterminate?language=objc) didn't seem entirely clear, but I guess it's saying it will always be called:

> For apps that do not support background execution or are linked against iOS 3.x or earlier, this method is always called when the user quits the app. For apps that support background execution, this method is generally not called when the user quits the app because the app simply moves to the background in that case. However, this method may be called in situations where the app is running in the background (not suspended) and the system needs to terminate it for some reason.

**Comment by bruno-garcia** [comment]
Oh I see. We also watch for when the app goes to the background to mark last time the session was active. I tested this event on the device and it was called when I closed the app, even after it was in the background.

File: Sources/Sentry/SentrySessionTracker.m [hunk 1]
```diff
@@ -72,4 +80,9 @@ - (void)willResignActive {
     self.lastInForeground = [NSDate date];
 }
 
+- (void)willTerminate {
+    NSDate *from = nil == self.lastInForeground ? [NSDate date] : self.lastInForeground;
```
Comment (philipphofmann) [inline Sources/Sentry/SentrySessionTracker.m:84]: `l`: Nice one-liner ^^. The naming for `from` is a little bit confusing for me. I don't really get it.

File: Sources/Sentry/SentrySessionTracker.m [hunk 2]
```diff
@@ -72,4 +80,9 @@ - (void)willResignActive {
...
+    [[SentrySDK currentHub] endSessionWithTimestamp:from];
+}
```
Comment (philipphofmann) [inline Sources/Sentry/SentrySessionTracker.m:86]: `m`: I know there aren't any tests right now for this class and there isn't much logic happening in this method, but still I would prefer having tests for `didBecomeActive`, `willResignActive` and `willTerminate`. Maybe we can split the Sentry logic from the `NSNotificationCenter` logic to make our code testable.

File: Sources/Sentry/SentrySessionTracker.m (hunk 1 above)
Comment (bruno-garcia) [reply Sources/Sentry/SentrySessionTracker.m:84]: Fair. It was the name taken before where there's some `from` `to` calculation.
I've renamed it here: https://github.com/getsentry/sentry-cocoa/pull/506

File: Sources/Sentry/SentrySessionTracker.m (hunk 2 above)
Comment (bruno-garcia) [reply Sources/Sentry/SentrySessionTracker.m:86]: Yeah need to dig into how to test those but generally need to test sessions, this class has 0 tests :hides:

Comment (bruno-garcia) [reply Sources/Sentry/SentrySessionTracker.m:86]: Session tests were added to pbxproj (got lost when moving to `xcworkspace`: https://github.com/getsentry/sentry-cocoa/pull/508

Still need to test the Tracker though and use the new DateTimeProvider bits

**Pull request merged**